spec/features/skip_rule_form_spec.rb

```diff
@@ -4,37 +4,186 @@
+      end
+    end
+
+    context "existing questioning" do
+      let(:dest_qing_str) { "Question ##{form.c[4].full_dotted_rank} #{form.c[4].code}" }
+      let(:first_cond_str) { "Question ##{form.c[0].full_dotted_rank} #{form.c[0].code} is less than 1000" }
+      let(:second_cond_str) { "Question ##{form.c[1].full_dotted_rank} #{form.c[1].code} is equal to 10" }
+
+      scenario "rules are updated correctly" do
+        visit("/en/m/#{form.mission.compact_name}/questionings/#{form.c[2].id}/edit")
+
+        # edit first rule
+        within(all(".skip-rule")[0]) do
+          find('select[name*="\\[skip_if\\]"]').select("if all of these conditions are met")
+          within(all(".condition-fields")[0]) do
+            select_question(form.c[0].code)
+            select_operator("< less than")
+            fill_in_value("1000")
+          end
+        end
+
+        # delete second rule
+        within(".rule-2") do
+          click_delete_link
+        end
+
+        click_on "Save"
+
+        rule_2 = "Skip to #{dest_qing_str} if all of these conditions are met "\
+        "#{first_cond_str} "\
+        "#{second_cond_str}"
+
+        visit("/en/m/#{form.mission.compact_name}/questionings/#{form.c[2].id}")
+        expect(page).to have_content(rule_2)
+      end
+
+      scenario "delete existing condition if rule is set to always" do
+        visit("/en/m/#{form.mission.compact_name}/questionings/#{form.c[2].id}/edit")
+
+        # confirm that data is available on visit
+        expect(page).to have_css(".condition-fields")
+        expect(page).to have_content("= equals")
+        expect(page).to have_select("questioning_skip_logic",
+          selected: "After this question, skip ...")
+
+        # set skip rule to always
+        within(all(".skip-rule")[0]) do
+          find('select[name*="\\[skip_if\\]"]').select("in all cases")
+
+          # condition fields are hidden for the first skip rule
+          expect(all(".condition-fields").size).to eq 0
+        end
+
+        # condition fields still exist for the second skip rule
+        within(all(".skip-rule")[1]) do
+          expect(all(".condition-fields").size).to eq 2
+        end
+
+      end
+
+      scenario "delete existing rule if skip is set to always" do
+        visit("/en/m/#{form.mission.compact_name}/questionings/#{form.c[2].id}/edit")
+
+        # confirm that data is available on visit
+        expect(page).to have_css(".condition-fields")
+        expect(page).to have_content("= equals")
+        expect(page).to have_select("questioning_skip_logic",
+          selected: "After this question, skip ...")
+
+        # set skip to always
+        select "After this question, go to the next question", from: "questioning_skip_logic"
+
+        expect(page).not_to have_css(".condition-fields")
+        expect(page).not_to have_content("= equals")
+        expect(page).not_to have_select("questioning_skip_logic",
+          selected: "After this question, skip ...")
```

Comment:
Typically it would be good to check that these changes are persisted properly as well (you did this in some of the earlier ones too), but I think this is fine for now.